cmd/benchmark.go

```diff
@@ -0,0 +1,320 @@
+		!strings.HasPrefix(pieces[1], "bpf=") {
+		return 0, 0, errors.New("invalid benchmark source string")
+	}
+	pieces[0] = strings.Split(pieces[0], "=")[1]
+	pieces[1] = strings.Split(pieces[1], "=")[1]
+	fc, err := strconv.ParseUint(pieces[0], 10, 64)
+	if err != nil {
+		return 0, 0, err
+	}
+	bpf, err := strconv.ParseInt(pieces[1], 10, 64)
+	if err != nil {
+		return 0, 0, err
+	}
+	return uint(fc), bpf, nil
+}
+
+var benchCmd *cobra.Command
+
+func init() {
+	raw := rawBenchmarkCmdArgs{}
+
+	// benCmd represents the bench command
+	benchCmd = &cobra.Command{
+		Use:        "bench [destination]",
+		Aliases:    []string{"ben", "benchmark"},
+		SuggestFor: []string{"b", "bn"},
+		Short:      benchCmdShortDescription,
+		Long:       benchCmdLongDescription,
+		Example:    benchCmdExample,
+		Args: func(cmd *cobra.Command, args []string) error {
+
+			// TODO: if/when we support benchmarking for S2S, note that the current code to set userAgent string in
+			//   jobPartMgr will need to be changed if we want it to still set the benchmarking suffix for S2S
+			if len(args) == 1 {
+				raw.dst = args[0]
+			} else {
+				return errors.New("wrong number of arguments, please refer to the help page on usage of this command")
+			}
+			return nil
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			var cooked cookedCopyCmdArgs // benchmark args cook into copy args
+			cooked, err := raw.cook()
+			if err != nil {
+				glcm.Error("failed to parse user input due to error: " + err.Error())
+			}
+
+			glcm.Info("Scanning...")
+
+			cooked.commandString = copyHandlerUtil{}.ConstructCommandStringFromArgs()
+			err = cooked.process()
+			if err != nil {
+				glcm.Error("failed to perform benchmark command due to error: " + err.Error())
+			}
+
+			glcm.SurrenderControl()
+		},
+	}
+	rootCmd.AddCommand(benchCmd)
+
+	benchCmd.PersistentFlags().StringVar(&raw.sizePerFile, common.SizePerFileParam, "250M", "size of each auto-generated data file. Must be "+sizeStringDescription)
+	benchCmd.PersistentFlags().UintVar(&raw.fileCount, common.FileCountParam, 100, "number of auto-generated data files to use")
+
+	benchCmd.PersistentFlags().Float64Var(&raw.blockSizeMB, "block-size-mb", 0, "use this block size (specified in MiB). Default is automatically calculated based on file size. Decimal fractions are allowed - e.g. 0.25. Identical to the same-named parameter in the copy command")
+	benchCmd.PersistentFlags().StringVar(&raw.blobType, "blob-type", "None", "defines the type of blob at the destination. Used to allow benchmarking different blob types. Identical to the same-named parameter in the copy command")
+	benchCmd.PersistentFlags().BoolVar(&raw.putMd5, "put-md5", false, "create an MD5 hash of each file, and save the hash as the Content-MD5 property of the destination blob/file. (By default the hash is NOT created.) Identical to the same-named parameter in the copy command")
+	// TODO use constant for default value or, better, move loglevel param to root cmd?
```

Comment:
Moving to root cmd would make sense, imo. logging is a pretty global thing in AzCopy.

Reply:
Yeah, will leave as TODO for now, since its currently at child level everywhere, and I don't have time to fix (and test) everywhere.  Will leave the TODO there to remind us in the future.